source/elements/oneMKL/source/domains/data_fitting/interpolate.rst

```diff
@@ -0,0 +1,106 @@
+.. _interpolate:
+
+Interpolate Function
+====================
+
+.. contents::
+    :local:
+    :depth: 1
+
+Interpolate function performs computations of function and derivatives values at interpolation sites.
+
+If the sites do not belong to interpolation interval ``[a, b]``, the library uses:
+
+  - interpolant :math:`I_0` coefficients computed for interval :math:`[x_0, x_1)` for the
+    computations at the sites to the left of ``a``.
+  - interpolant :math:`I_{n-2}` coefficients computed for interval
+    :math:`[x_{n-2}, x_{n-1})` for the computations at the sites to the right of ``b``.
+
+Interpolation algorithm depends on interpolant's type (e.g., for cubic spline
+interpoilation evaluation of third-order polynomial is performed to obtain function values).
+
+Header File
+-----------
+
+.. code:: cpp
+
+  #include<oneapi/mkl/experimental/data_fitting.hpp>
+
+Namespace
+---------
+
+.. code:: cpp
+
+  oneapi::mkl::experimental::data_fitiing
+
+Syntax
+------
+
+.. code:: cpp
+
+  template <typename Interpolant>
+  sycl::event interpolate(
+      Interpolant& interpolant,
```

Comment:
are these missing the queue as argument?

Also, do you fit into non-member or member function cases as seen in https://github.com/oneapi-src/oneAPI-spec/blob/main/source/elements/oneMKL/source/architecture/execution_model.inc.rst#member-functions ?  If you are of member-function approach, you may want to add some language there to include Data fitting in list with DFt and RNG.

Reply:
What about changing the spec to say something like this:

> In general, each oneMKL non-member computational routine takes a sycl::queue reference as its first parameter. 
 In some exceptional cases, like in Data Fitting, it may instead take an object as first parameter that itself contains a reference to the sycl execution context.

Reply:
why is a queue not passed in?  why is a context sufficient ?  Is there a future call which adds the queue ?   How does work get done ?

Reply:
Actually the queue is inside (See here: https://github.com/oneapi-src/oneAPI-spec/pull/413/files/#diff-2361f70703f03fec4684fd287b54dae6e0b303c2f095bab62be9b5bc536b51f8R34-R41).
Sorry, I might misspell it trying to explain it more generally.